### ISSUE TITLE
XCode 10 GM - Crash on Tapping push notifications 

### ISSUE THREAD

**Issue by Roshan-SignEasy** [mixpanel/mixpanel-swift#237]
Hi ,

We are getting the following crash on tapping any push notification in our app. [Stack trace]

**$SSo8NSObjectC8MixpanelE22userNotificationCenter_13newDidReceive21withCompletionHandlerySo06UNUserdE0C_So22UNNotificationResponseCyyctF (AutomaticEvents.swift:289)**

When you receive a push notification, the app opens and crashes.

This issue was not present when we build using Xcode 9

Thanks

**Comment by Roshan-SignEasy** [comment]
<img width="1421" alt="screen shot 2018-09-17 at 5 20 57 pm" src="https://user-images.githubusercontent.com/38778308/45621488-1bdf0580-ba9e-11e8-963d-d0fdf15f6da2.png">
More info here.

**Comment by zihejia** [comment]
hi @Roshan-SignEasy , thanks for raising it. We have been working on the XCode 10 support now, so this issue will be taken care of. 
This issue might be related:
https://github.com/mixpanel/mixpanel-swift/issues/232

**Comment by zihejia** [comment]
We have released 2.5.0 which supports Xcode 10. The issue could not be reproduced, however if the problem still exists, please reopen it.

**Comment by samwarfield** [comment]
I am experiencing this same issue

```
Thread 1: closure argument passed as @noescape to Objective-C has escaped: 
file /mixpanel-swift/Mixpanel/AutomaticEvents.swift, line 289, column 77
```

<img width="721" alt="screen shot 2018-09-20 at 4 21 57 pm" src="https://user-images.githubusercontent.com/5303872/45851807-96539380-bcf1-11e8-843a-a39785fc5d84.png">


```
0    Mixpanel                           0x0000000104ccfa30 NSObject.userNotificationCenter(_:newDidReceive:withCompletionHandler:) + 648
1    Mixpanel                           0x0000000104ccff6c @objc NSObject.userNotificationCenter(_:newDidReceive:withCompletionHandler:) + 124
2    UIKitCore                          0x00000001c3704f48 <redacted> + 3444
3    UIKitCore                          0x00000001c3f14fa0 <redacted> + 2432
4    UIKitCore                          0x00000001c3ed42f8 <redacted> + 772
5    UIKitCore                          0x00000001c3ed40f8 <redacted> + 432
6    UIKitCore                          0x00000001c3f17768 <redacted> + 220
7    UIKitCore                          0x00000001c3f182c4 _performActionsWithDelayForTransitionContext + 112
8    UIKitCore                          0x00000001c3f17604 <redacted> + 248
9    UIKitCore                          0x00000001c3f108a0 <redacted> + 368
10   UIKitCore                          0x00000001c371d76c <redacted> + 468
11   FrontBoardServices                 0x000000019986c4bc <redacted> + 228
12   libdispatch.dylib                  0x00000001080acdd4 _dispatch_client_callout + 16
13   libdispatch.dylib                  0x00000001080b0944 _dispatch_block_invoke_direct + 232
14   FrontBoardServices                 0x00000001998aa618 <redacted> + 40
15   FrontBoardServices                 0x00000001998aa12c <redacted> + 416
16   FrontBoardServices                 0x00000001998aa8b0 <redacted> + 56
17   CoreFoundation                     0x0000000196dca5a0 <redacted> + 24
18   CoreFoundation                     0x0000000196dca4e0 <redacted> + 88
19   CoreFoundation                     0x0000000196dc9d6c <redacted> + 176
20   CoreFoundation                     0x0000000196dc48d8 <redacted> + 1040
21   CoreFoundation                     0x0000000196dc4404 CFRunLoopRunSpecific + 436
22   GraphicsServices                   0x0000000199038520 GSEventRunModal + 100
23   UIKitCore                          0x00000001c36ef484 UIApplicationMain + 212
```
